File: zigar-runtime/test/doc-examples/add-error.js
import { add, MathError } from './add-error.zig';

console.log(add(1, 2));
try {
  add(0, 1);
} catch (err) {
  console.error(err);
  console.log(err instanceof MathError);
}

// console output:
// 3
// [ErrorSet0000 [Error]: Unexpected spanish inquisition]
// true

File: zigar-runtime/test/doc-examples/add-error.zig
// add-error.zig
pub const MathError = error{
    UnexpectedSpanishInquisition,
    RecordIsScratched,
};

pub fn add(a: i32, b: i32) !i32 {
    if (a == 0 or b == 0) {
        return MathError.UnexpectedSpanishInquisition;
    }
    return a + b;
}
